receiver/oracledbreceiver/db_client.go

```diff
@@ -0,0 +1,99 @@
+// Copyright Splunk, Inc.
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oracledbreceiver // import "github.com/signalfx/splunk-otel-collector/receiver/oracledbreceiver"
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"reflect"
+	"strings"
+
+	// register db driver
+	_ "github.com/sijms/go-ora/v2"
+	"go.uber.org/zap"
+)
+
+type dbClient interface {
+	metricRows(ctx context.Context) ([]metricRow, error)
+}
+
+type metricRow map[string]string
+
+type dbSQLClient struct {
+	db     *sql.DB
+	logger *zap.Logger
+	sql    string
+}
+
+func newDbClient(db *sql.DB, sql string, logger *zap.Logger) dbClient {
+	return dbSQLClient{
+		db:     db,
+		sql:    sql,
+		logger: logger,
+	}
+}
+
+func (cl dbSQLClient) metricRows(ctx context.Context) ([]metricRow, error) {
```

Comment:
This looks like https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/sqlqueryreceiver/db_client.go was c/p instead of wrapping the sqlquery receiver and configuring w/ default queries. Creating diverging implementations out of the gate seems undesirable.

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14321 could be adopted after the fact.